app/lib/order_txt.rb

```diff
@@ -17,11 +19,39 @@ def to_txt
     text += '****** ' + I18n.t('orders.fax.to_address') + "\n\n"
     text += "#{FoodsoftConfig[:name]}\n#{contact[:street]}\n#{contact[:zip_code]} #{contact[:city]}\n\n"
     text += '****** ' + I18n.t('orders.fax.articles') + "\n\n"
-    text += format("%8s %8s   %s\n", I18n.t('orders.fax.number'), I18n.t('orders.fax.amount'),
-                   I18n.t('orders.fax.name'))
+
+    # prepare order_articles data
+    longest_number_string_length = 0
+    longest_amount_string_length = I18n.t('orders.fax.amount').length
+    longest_unit_string_length = I18n.t('orders.fax.unit').length
+    @order_positions = @order.order_articles.ordered.includes(:article_version).map do |oa|
+      number = oa.article_version.order_number || ''
+      amount = format_units_to_order(oa).to_s
+      unit = format_supplier_order_unit_with_ratios(oa.price)
+      longest_number_string_length = number.length if number.length > longest_number_string_length
+      longest_amount_string_length = amount.length if amount.length > longest_amount_string_length
+      longest_unit_string_length = unit.length if unit.length > longest_unit_string_length
+      {
+        number: number,
+        amount: amount,
+        unit: unit,
+        name: oa.article_version.name
+      }
+    end
+
+    if (any_number_present = longest_number_string_length > 0) && longest_number_string_length < I18n.t('orders.fax.number').length
+      longest_number_string_length = I18n.t('orders.fax.number').length
+    end
```

Comment:
This would do the same with a little less redundancy:

```suggestion
    order_articles =  @order.order_articles.ordered.includes(:article_version)
    order_positions = order_articles.map do |oa|
      number = oa.article_version.order_number || ''
      amount = format_units_to_order(oa).to_s
      unit = format_supplier_order_unit_with_ratios(oa.price)
      {
        number: number,
        amount: amount,
        unit: unit,
        name: oa.article_version.name
      }
    end

    any_number_present = order_articles.any? { |oa| oa.article_version.order_number.present? }
    longest_number_string_length = (order_positions.pluck(:number) + [I18n.t('orders.fax.number')]).map(&:length).max if any_number_present
    longest_amount_string_length = (order_positions.pluck(:amount) + [I18n.t('orders.fax.amount')]).map(&:length).max
    longest_unit_string_length = (order_positions.pluck(:unit) + [I18n.t('orders.fax.unit')]).map(&:length).max
```

It would be even better to make this more generic though, so we wouldn't have to alter the code so much whenever we remove/add a column and to reduce the method's size (too big after the changes!).
This could be done by using something like https://github.com/aptinio/text-table (<- quite antique, so either we find something newer or do our own implementation.)

All this is quite out of scope as far as the hackathon fork is concerned though, and I wonder if we should implement dynamic column widths here at all. (Should probably be discussed upstream instead.)